systems/introduction/vm.go

```diff
@@ -28,16 +30,47 @@ const (
 func compute(memory []byte) {
 
 	registers := [3]byte{8, 0, 0} // PC, R1 and R2
+	var pcincr byte               // Amount to increment the PC each cycle
 
 	// Keep looping, like a physical computer's clock
 	for {
+		pcincr = 3 // Default instruction length is three bytes
+
+		op := memory[registers[0]]
+		arg1 := memory[registers[0]+1]
+		arg2 := memory[registers[0]+2]
 
-		// op := TODO // fetch the opcode
+		// decode and execute
+		switch op {
+		case Load:
+			registers[arg1] = memory[arg2]
+		case Store:
+			if arg2 >= 8 {
```

Comment:
May be nit-picking, but `8` may be a magic number unless you know what it indicates. If you agree, then you can extract it to a constant.